### PR TITLE
Fixes HttpClient Content.Source reads from arbitrary threads

### PR DESCRIPTION
**Pull request by sbordet**

### CODE REVIEW

**Comment by lorban** [comment]
Is that an alternative for the fix in #12143?

**Review by lorban — CHANGES_REQUESTED** [review]
I don't mind this change, as long as the `SerializedInvoker` assertions added in #12143 don't trip. So I think they should be added as part of this PR too.

File: jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpReceiver.java
Comment (lorban) [inline jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpReceiver.java:1]: This class should have the `SerializedInvoker` assertions of https://github.com/jetty/jetty.project/pull/12143

**Review by lorban — CHANGES_REQUESTED** [review]
This is much cleaner than https://github.com/jetty/jetty.project/pull/12143. After modifying FCGI too, I can't insist enough that the `SerializedInvoker` assertions should be added too.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java [hunk 1]
```diff
@@ -116,8 +153,8 @@ public void run(Runnable task)
         if (todo != null)
             todo.run();
         else
-            if (LOG.isDebugEnabled())
-                LOG.debug("Queued link in {}", this);
+        if (LOG.isDebugEnabled())
```
Comment (sbordet) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java]: `else if` on the same line.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java [hunk 2]
```diff
@@ -131,22 +168,22 @@ public void run(Runnable... tasks)
         if (todo != null)
             todo.run();
         else
-            if (LOG.isDebugEnabled())
-                LOG.debug("Queued links in {}", this);
+        if (LOG.isDebugEnabled())
```
Comment (sbordet) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java]: Ditto.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java [hunk 3]
```diff
@@ -209,10 +271,35 @@ public String toString()
...
-    private record NamedRunnable(Runnable delegate, String name) implements Runnable
+    private class NamedRunnable implements Runnable
     {
         private static final Logger LOG = LoggerFactory.getLogger(NamedRunnable.class);
 
+        private final Runnable delegate;
+        private final String name;
+        private final Throwable stack;
+
+        public NamedRunnable(Runnable delegate)
```
Comment (sbordet) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java]: private

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java [hunk 4]
```diff
@@ -209,10 +271,35 @@ public String toString()
...
+        {
+            this.delegate = delegate;
+            this.stack = new Throwable();
+            this.name = deriveTaskName(delegate, stack);
+        }
+
+        protected String deriveTaskName(Runnable task, Throwable stack)
```
Comment (sbordet) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java]: private

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java (hunk 1 above)
Comment (lorban) [reply jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java]: done.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java (hunk 2 above)
Comment (lorban) [reply jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java]: done.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java (hunk 3 above)
Comment (lorban) [reply jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java]: done.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java (hunk 4 above)
Comment (lorban) [reply jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java]: done.

**Comment by lorban** [comment]
@sbordet nudge

**Review by lorban — APPROVED** [review]

**Pull request merged**